cmake/CPM.cmake

```diff
@@ -251,7 +251,14 @@ function(CPMFindPackage)
     endif()
   endif()
 
-  if(CPM_DOWNLOAD_ALL)
+  set(download_only ${CPM_DOWNLOAD_ALL})
+  if(DEFINED ENV{CPM_${CPM_ARGS_NAME}_DOWNLOAD})
+    set(download_only $ENV{CPM_${CPM_ARGS_NAME}_DOWNLOAD})
+  endif()
+  if(DEFINED CPM_${CPM_ARGS_NAME}_DOWNLOAD)
+    set(download_only ${CPM_${CPM_ARGS_NAME}_DOWNLOAD})
+  endif()
```

Comment:
kinda nitpicky but since local variable trumps env which trumps global perhaps just do:

```cmake
if(DEFINED local)
  # ...
elseif(DEFINED env)
  # ...
else()
  set(download_only ${CPM_DOWNLOAD_ALL})
endif()
```
This makes the priority clearer, no?

Reply:
Yes that is better.